sql/src/main/java/io/crate/expression/scalar/arithmetic/ArithmeticFunctions.java

```diff
@@ -208,4 +254,89 @@ public static Function of(String name, Symbol first, Symbol second, Set<Function
             ImmutableList.of(first, second)
         );
     }
+
+    private static final class IntervalArithmeticScalar extends Scalar<Period, Object> {
+
+        private final FunctionInfo info;
+        private final BiFunction<Period, Period, Period> operation;
+
+        IntervalArithmeticScalar(String operator) {
+            this.info = new FunctionInfo(
+                new FunctionIdent("add", Arrays.asList(DataTypes.INTERVAL, DataTypes.INTERVAL)), DataTypes.INTERVAL);
+            switch (operator) {
+                case "+":
+                    operation = Period::plus;
+                    break;
+                case "-":
+                    operation = Period::minus;
+                    break;
+                default:
+                    throw new IllegalArgumentException("Unsupported operator for interval " + operator);
+            }
+        }
+
+        @Override
+        public Period evaluate(TransactionContext txnCtx, Input<Object>... args) {
+            Period fst = (Period) args[0].value();
+            Period snd = (Period) args[1].value();
+
+            if (fst == null || snd == null) {
+                return null;
+            }
+            return operation.apply(fst, snd);
+        }
+
+        @Override
+        public FunctionInfo info() {
+            return this.info;
+        }
+    }
+
+    private static final class IntervalTimestampScalar extends Scalar<Long, Object> {
+
+        private final BiFunction<DateTime, Period, DateTime> operation;
+        private final FunctionInfo info;
+
+        IntervalTimestampScalar(String operator, DataType firstType, DataType secondType, DataType returnType) {
+            this.info = new FunctionInfo(new FunctionIdent("add", Arrays.asList(firstType, secondType)), returnType);
+            switch (operator) {
+                case "+":
+                    operation = DateTime::plus;
+                    break;
+                case "-":
```

Comment:
I missed this earlier, but `interval - timestamp` doesn't make sense. (In PostgreSQL it is also unsupported)